packages/quantic/force-app/main/default/lwc/quanticPager/e2e/quantic-pager.e2e.ts

```diff
@@ -0,0 +1,218 @@
+  await page.goto(`${pageUrl}#${urlHash}`);
+  await search.interceptSearchAndLimitResultPages(5);
+  configuration.configure();
+}
+
+async function setupWithPauseBeforeSearch(
+  page: Page,
+  {search, insight, configuration}: PageObjects,
+  options: Partial<PagerOptions> = {}
+) {
+  const pageUrl = 's/quantic-pager';
+  await page.goto(pageUrl);
+
+  if (options.useCase === useCaseEnum.insight) {
+    configuration.configure(options);
+    await insight?.waitForInsightInterfaceInitialization();
+    const stalledRequest = search.interceptSearchIndefinitely();
+    await search.performSearch();
+    return stalledRequest;
+  }
+  const stalledRequest = search.interceptSearchIndefinitely();
+  configuration.configure(options);
+  return stalledRequest;
+}
+
+const fixtures = {
+  search: testSearch,
+  insight: testInsight,
+};
+
+useCaseTestCases.forEach((useCase) => {
+  let test = fixtures[useCase.value];
+
+  test.describe(`quantic pager ${useCase.label}`, () => {
+    test.beforeEach(async ({page, search, insight, configuration}) => {
+      await visitPage(
+        page,
+        {search, insight, configuration},
+        {useCase: useCase.value}
+      );
+    });
+
+    test.describe('when the page is loading ${useCase.label', () => {
+      test('should display the pager only after the search request has been completed', async ({
+        page,
+        search,
+        insight,
+        pager,
+        configuration,
+      }) => {
+        const resumeSearchRequest = await setupWithPauseBeforeSearch(
+          page,
+          {search, insight, configuration},
+          {useCase: useCase.value}
+        );
+
+        await expect(pager.previousPageButton).not.toBeVisible();
+        await expect(pager.nextPageButton).not.toBeVisible();
+        await expect(pager.pageButtons).not.toBeVisible();
+
+        resumeSearchRequest();
+
+        await expect(pager.previousPageButton).toBeVisible();
+        await expect(pager.nextPageButton).toBeVisible();
+        await expect(pager.pageButtons).not.toBeNull();
+      });
+    });
+
+    test.describe('when being in the first page', () => {
+      test('should disable the previous page button', async ({pager}) => {
+        await expect(pager.pageButtons.nth(0)).toHaveAttribute(
+          'aria-pressed',
+          'true'
+        );
+        await expect(pager.previousPageButton).toBeDisabled();
+      });
+    });
+
+    test.describe('when being in the last page', () => {
+      test.beforeEach(async ({pager, search}) => {
+        await pager.goToLastPage();
+        await search.waitForSearchResponse();
+      });
+
+      test('should disable the next page button', async ({pager}) => {
+        await expect(pager.pageButtons.nth(-1)).toHaveAttribute(
+          'aria-pressed',
+          'true'
+        );
+        await expect(pager.nextPageButton).toBeDisabled();
+      });
+    });
+
+    test.describe('when clicking the next page button', () => {
```

Comment:
In those "navigation" tests, you could also check the expected page in the response.

Reply:
This is exactly spot-on what should be e2e tested IMO.

A user interaction with the component (click) that is supposed to trigger network requests and/or a change in the page/url/something browser releated such as localStorage/cookies/etc.

And completely agree with @fbeaudoincoveo this is the place where we should be testing the "full loop" so clicking on the pager means 

- a new search request (potentially testing also in the request to check the value of `firstResult` too)
- a new analytics request (testing the event type, body, being sent, and ultimately the response too, even moreso for EP when we'll have the /validate (out of scope for this PR IMO))
- a new search response (testing for the `firstResult` in the response too)
- also potentially changes in the url? that we now have `&firstResult=10` in the url, but only for the search use-case not insight?
- maybe also testing some aria-live stuff/accessibility too when the time will be there.